go/libraries/doltcore/schema/typeinfo/float32.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2019 Liquidata, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package typeinfo
+
+import (
+	"fmt"
+	"math"
+
+	"github.com/src-d/go-mysql-server/sql"
+
+	"github.com/liquidata-inc/dolt/go/store/types"
+)
+
+type float32Impl struct{}
+
+var _ TypeInfo = (*float32Impl)(nil)
+
+// AppliesToValue implements TypeInfo interface.
+func (ti *float32Impl) AppliesToKind(kind types.NomsKind) error {
+	if kind != types.FloatKind {
+		return fmt.Errorf(`kind "%v" is not applicable to typeinfo "%v"`, kind.String(), ti.String())
+	}
+	return nil
+}
+
+// ConvertNomsValueToValue implements TypeInfo interface.
+func (ti *float32Impl) ConvertNomsValueToValue(v types.Value) (interface{}, error) {
+	if val, ok := v.(types.Float); ok {
+		return float32(val), nil
+	}
+	return nil, fmt.Errorf(`"%v" cannot convert NomsKind "%v" to a value`, ti.String(), v.Kind())
+}
+
+// ConvertValueToNomsValue implements TypeInfo interface.
+func (ti *float32Impl) ConvertValueToNomsValue(v interface{}) (types.Value, error) {
+	return floatConvertValueToNomsValue(ti, v)
+}
+
+// Equals implements TypeInfo interface.
+func (ti *float32Impl) Equals(other TypeInfo) bool {
+	if other == nil {
+		return false
+	}
+	if ti2, ok := other.(*float32Impl); ok {
+		return *ti == *ti2
```

Comment:
This is always true (empty structs are always == to each other. Seems like you just want to do a type assertion here.

Reply:
Same comment applies to all of these types